File: windowslib/src/main/resources/scala-native/windows/winnt/sidNameUse.c
#if defined(_WIN32) || defined(WIN32)
#define WIN32_LEAN_AND_MEAN
#include <windows.h>
#include <winnt.h>

int scalanative_sidtypeuser() { return SidTypeUser; }
int scalanative_sidtypegroup() { return SidTypeGroup; }
int scalanative_sidtypedomain() { return SidTypeDomain; }
int scalanative_sidtypealias() { return SidTypeAlias; }
int scalanative_sidtypewellknowngroup() { return SidTypeWellKnownGroup; }
int scalanative_sidtypedeletedaccount() { return SidTypeDeletedAccount; }
int scalanative_sidtypeinvalid() { return SidTypeInvalid; }
int scalanative_sidtypeunknown() { return SidTypeUnknown; }
int scalanative_sidtypecomputer() { return SidTypeComputer; }
int scalanative_sidtypelabel() { return SidTypeLabel; }
int scalanative_sidtypelogonsession() { return SidTypeLogonSession; }
#endif // defined(Win32)

File: windowslib/src/main/scala/scala/scalanative/windows/winnt/SidNameUse.scala
package scala.scalanative.windows.winnt

import scalanative.unsafe.*
@extern
object SidNameUse {
  @name("scalanative_sidtypeuser")
  def SidTypeUser: SidNameUse = extern

  @name("scalanative_sidtypegroup")
  def SidTypeGroup: SidNameUse = extern

  @name("scalanative_sidtypedomain")
  def SidTypeDomain: SidNameUse = extern

  @name("scalanative_sidtypealias")
  def SidTypeAlias: SidNameUse = extern

  @name("scalanative_sidtypewellknowngroup")
  def SidTypeWellKnownGroup: SidNameUse = extern

  @name("scalanative_sidtypedeletedaccount")
  def SidTypeDeletedAccount: SidNameUse = extern

  @name("scalanative_sidtypeinvalid")
  def SidTypeInvalid: SidNameUse = extern

  @name("scalanative_sidtypeunknown")
  def SidTypeUnknown: SidNameUse = extern

  @name("scalanative_sidtypecomputer")
  def SidTypeComputer: SidNameUse = extern

  @name("scalanative_sidtypelabel")
  def SidTypeLabel: SidNameUse = extern

  @name("scalanative_sidtypelogonsession")
  def SidTypeLogonSession: SidNameUse = extern
}
